### PR TITLE
Enabling serialization for CultureInfo.

### PR DESCRIPTION
**Pull request by SGuyGe**
@shmao @zhenlan @khdang

### CODE REVIEW

File: src/System.Private.DataContractSerialization/src/System/Runtime/Serialization/ClassDataContract.cs [hunk 1]
```diff
@@ -456,7 +456,11 @@ static internal bool IsNonAttributedTypeValidForSerialization(Type type)
             { "System.Tuple`8", Array.Empty<string>() },
             { "System.Collections.Queue", new [] {"_syncRoot" } },
             { "System.Collections.Stack", new [] {"_syncRoot" } },
-            { "System.Globalization.CultureInfo", Array.Empty<string>() },
+            { "System.Globalization.CultureInfo", new [] { "m_cultureData", "m_isInherited", "m_nonSortName", "m_parent", "m_sortName" } },
+            { "System.Globalization.CompareInfo", new[] { "m_dataHandle", "m_handleOrigin", "m_sortName" } },
```
Comment (ellismg) [inline src/System.Private.DataContractSerialization/src/System/Runtime/Serialization/ClassDataContract.cs]: Can you help me understand what the significance of the field names in this array are?  I worry about m_dataHandle and m_handleOrigin being persisted.

Comment (SGuyGe) [inline src/System.Private.DataContractSerialization/src/System/Runtime/Serialization/ClassDataContract.cs]: The fields listed are to be EXCLUDED from serialization, this is the alternative to [NonSerialized] attribute.

**Comment by SGuyGe** [comment]
@dotnet-bot please test

**Comment by stephentoub** [comment]
@dotnet-bot test this please

File: src/System.Private.DataContractSerialization/src/System/Runtime/Serialization/ClassDataContract.cs (hunk 1 above)
Comment (ellismg) [inline src/System.Private.DataContractSerialization/src/System/Runtime/Serialization/ClassDataContract.cs]: Great!

**Comment by shmao** [comment]
I'm closing the PR as tests failed with it. I will take the PR and make some changes.

**Pull request closed without merging**